Chore/feedback implementation

Celoka opened the pull request.

Celoka committed:
Merge pull request #3 from Celoka/routes

Routes

Celoka committed:
Merge pull request #4 from Celoka/routes

Code refactor

Celoka committed:
Merge pull request #5 from Celoka/routes

Edited the addUserToGroup route to display the name of group members

Celoka committed:
Merge pull request #7 from Celoka/routes

Changed firebase configuration key

Celoka committed:
Merge pull request #8 from Celoka/routes

config change

Celoka committed:
Create README.md

Celoka committed:
Update README.md

Celoka committed:
Update README.md

Celoka committed:
Update README.md

Celoka committed:
Merge pull request #9 from Celoka/routes

Code Refactor

Celoka committed:
API unit test

Celoka committed:
Merge pull request #11 from Celoka/routes

API unit test

Celoka committed:
API Unit Test: Server and test file.

Celoka committed:
Merge pull request #12 from Celoka/routes

API Unit Test: Server and Test file.

Celoka committed:
API Route and configuration files

Celoka committed:
Merge pull request #13 from Celoka/routes

API Route and configuration files

Celoka committed:
Delete server.js

Celoka committed:
Delete config.js

Celoka committed:
Delete routes.js

Celoka committed:
Update README.md

Celoka committed:
Merge pull request #14 from Celoka/develop1

Develop1

Celoka committed:
Merge pull request #15 from Celoka/routes

Merge pull request #14 from Celoka/develop1

Celoka committed:
Folder restructure

Celoka committed:
Folder restructuring

Celoka committed:
Merge pull request #16 from Celoka/routes

Routes

Celoka committed:
Wrote more test for routes

Celoka committed:
Modified the file: test.js is now served by server.js. Deleted route-testserver.js

Celoka committed:
Merge pull request #17 from Celoka/routes

Wrote more test for routes

Celoka committed:
Created new files for post-it app

Celoka committed:
Merge pull request #18 from Celoka/routes

Created new files for post-it app

Celoka committed:
General code refactor

Celoka committed:
Merge pull request #19 from Celoka/routes

General code refactor

Celoka committed:
Test.js: Code Refactor

Celoka committed:
Merge pull request #20 from Celoka/routes

Test.js: Code Refactor

Celoka committed:
Folder restructure and creation of new components

Celoka committed:
Merge pull request #21 from Celoka/routes

Folder restructure and creation of new components

Celoka committed:
Delete appstore.js

Celoka committed:
Merge pull request #22 from Celoka/routes

Delete appstore.js

Celoka committed:
Deletion of one public folder in Client Directory

Celoka committed:
Merge pull request #23 from Celoka/routes

Deletion of one public folder in Client Directory

Celoka committed:
Edited all the components and modified the routes.

Celoka committed:
Merge pull request #24 from Celoka/routes

Edited all the components and modified the routes.

Celoka committed:
Modified LoginPAge and SignUpPage

Celoka committed:
Merge pull request #25 from Celoka/routes

Modified LoginPage and SignUpPage

Celoka committed:
App component Test

Celoka committed:
Merge pull request #26 from Celoka/routes

App component Test

Celoka committed:
Code Refactor.

Celoka committed:
Merge pull request #27 from Celoka/routes

Code Refactor.

Celoka committed:
Code Refactor: Changed all the application components.

Celoka committed:
Merge pull request #28 from Celoka/routes

Code Refactor: Changed all the application components.

Celoka committed:
Modified Addmember, Creategroup and Footer components. Deleted app folder in dist

Celoka committed:
Merge pull request #29 from Celoka/routes

Modified Addmember, Creategroup and Footer components. Deleted app fo…

Celoka committed:
Delete HomePage.jsx

Celoka committed:
File modification

Celoka committed:
Deleted Homepage.jsx from Components folder

Celoka committed:
Delete HomePage.jsx

Celoka committed:
Delete HomePage.jsx

Celoka committed:
Code refcator

Celoka committed:
Resolved Conflict

Celoka committed:
Homepage.jsx Delete

Celoka committed:
Delete HomePage.jsx

Celoka committed:
Delete HomePage.jsx

Celoka committed:
Line Edit in Footer

Celoka committed:
Footer.jsx Edit

Celoka committed:
Merge pull request #32 from Celoka/routes

Routes

Celoka committed:
Code refactor

Celoka committed:
Merge pull request #33 from Celoka/routes

Code refactor

Celoka committed:
Modification of server, webpack server and package.json

Celoka committed:
Merge pull request #34 from Celoka/routes

Modification of server.js, webpack server and package.json

Celoka committed:
React hot reloader installed in dependecies

Celoka committed:
Refactored webpack config

Celoka committed:
Merge pull request #35 from Celoka/routes

React hot reloader installed in dependencies

Celoka committed:
Added JsDoc comment

Celoka committed:
Merge pull request #36 from Celoka/routes

Added JsDoc comment

Celoka committed:
Component Name change Sign Up And SignIn

Celoka committed:
Merge pull request #37 from Celoka/routes

Component Name change Sign Up And SignIn

Celoka committed:
Removed a Route duplicate for home component

Celoka committed:
Merge pull request #38 from Celoka/routes

Removed a Route duplicate for home component

Celoka committed:
Git ignored dist folder and created server folder for server and routes

Celoka committed:
Merge pull request #39 from Celoka/routes

Git ignored dist folder and created server folder for server and routes

Celoka committed:
feature: integrate  .env, create controllers for routes, refactor api.

-refactor route.
-create controllers for users and groups.
-create a donenv for firebase config key.

Celoka committed:
Merge pull request #40 from Celoka/routes

feature: integrate  .env, create controllers for routes, refactor api.

Celoka committed:
Route refactor

Celoka committed:
Merge pull request #41 from Celoka/routes

Route refactor

Celoka committed:
Reset Password

Celoka committed:
Merge pull request #42 from Celoka/routes

Reset Password

Celoka committed:
feature: implement post message to group

- create messageboard component
- create actions, dispatch actions, and message store

[Deliver #4]

Celoka committed:
feature(message): implement retrieve message sent to group

- create child component for message board and pass props to it
- pass props from dashboard to message board
- install nexmo and nodemailer
- refactor post message and get message controllers

Celoka committed:
Merge pull request #45 from Celoka/feature/retrieve-group-message

Feature/retrieve group message

Celoka committed:
Create LICENSE

Celoka committed:
Update README.md

Celoka committed:
feature(add user): implement add user to group

- add JSdoc to all the required functions

Celoka committed:
Merge pull request #47 from Celoka/feature/add-user-to-group

feature/add user to group

Celoka committed:
Merge pull request #48 from Celoka/routes

feature/ add user to group

Celoka committed:
chore(test):  write server side test

- write side test using chai to test the quality of the code written.
- installed all the test dependencies required.

Celoka committed:
chore(test):  write server side test

    - deleted auto generated files

Celoka committed:
feature(add-user): implement add user to group

- refactor add user route
- re-configure webpack to allow for google sign in implementation
- create a dotenv file to add secret key to the environment variable
- reconfigure app store and action to fix bug
- catch errors at the view
- set up front end test with jest and enzyme

[Delivers  #001 ]

Celoka committed:
feature(notification): implement message notification

- use nodemailer and nexmo to implement sms notification

[Delivers 001]

Celoka committed:
bug(flight): fix sms notification bug

 -refactor code for nexmo to send sms notification

[Delivers 001]

Celoka committed:
bug(flight): fix sms notification bug

 - refactor code in dashboard component and group component
- fix post message bug
- remove all console log

[Delivers 001]

Celoka committed:
feature(add-member): implement add registered member to group

- complete feature implementation
- change the view interface of the app

[Finishes ]

Celoka committed:
feature(cursor-pointer): implement cursor pointer

- add cursor pointer for on hover and click group

[Finishes ]

Celoka committed:
refactor(): ensure appriopriate component type are used

- refcatoring component code to change some stateful components to stateless

[Finishes]

Celoka committed:
refactor(env): ensure secret keys are put in environment variable

- refcatoring component code to change some stateful components to stateless

 [Finishes]

Celoka committed:
test(componen): write test for all components and actions

- write test for some component
- set up test for actions
- create webpack for production

[Finishes]

Celoka committed:
Merge pull request

Celoka committed:
feature(readme): update readme file

- edit read me to show build and coverage

[Finishes ]

Celoka committed:
refactor(): refactore code in message board component

- set group name to state after receiving as props from parent component

[Finihses ]

Celoka committed:
test(component): write more component test

- write more component test to increase coverage

[Finishes ]

Celoka committed:
documentation(jsdoc):  write jsdoc for all function

- include jsdoc for all actions and methods
- edit eslint and include rules to adhere properly to airbnb style guide

[Finishes ]

Celoka committed:
documentation(jsdoc-refactor): give better description

- included and changed conflicting information in jsdoc for some stateless component

[Finishes ]

Celoka committed:
chore(webpack): set up webpack

- set up webpack for production

[Finishes ]

Celoka committed:
chore(webpack): set up webpack

- edit production webpack

[Finishes ]

Celoka committed:
test(unit): adding missing test

- complete test for component
- write test for actions

[Finishes ]

Celoka committed:
test(unit): write and complete unit test

- complete component test
- complete action test

[Finishes ]

Celoka committed:
feat(ui): style dashboard

-  include icon for in-app notification
- align it with other elements on dashboard

[Finishes ]

Celoka committed:
chore(test): update server side test

- update server side to and complete test
- edit route to eliminat dead code
- edit webpack for production full option

[Finishes ]

Celoka committed:
chore(test):  failing build

- edit travis.yml in the root folder for possible build

[Finishes ]

Celoka committed:
chore(test):  failing build

- edit travis.yml in the root folder for possible build

 [Finishes ]

Celoka committed:
chore(test):  loadmessage.json file not found in axios.js

- edit axios to solve problem

[Finishes ]

Celoka committed:
chore(docs): update read me

- paste build and coverage markdown to reflect current status and percentage respectively

[Finishes ]

Celoka committed:
chore(badge): add badge from code climate

- paste badge markdown from code climate in read me file

[Finishes ]

Celoka committed:
chore(install dependency): install babel cli dependency

- install babel cli to enable  some shell-executable utility script especially babel node for build success on heroku

[Finishes ]

Celoka committed:
chore(install dependency): install babel polyphill dependency

- install babel polyphill to server the serve to avoid using bable node on heroku

[Finishes ]

Celoka committed:
chore(webpack):  production webpack reconfiguration

- include conditional statement in server to bundle files with production webpack
- include mit licence markdown in read me file
- refactor start script in package.json

[Finishes ]

Celoka committed:
feature(login): implement google signin

- create a controller to handle Id token sent from the client side

Celoka committed:
feature(): remove message icon

- remove message icon

Celoka committed:
chore(refactor): solve indentations issues

- solve indentation issues highlighted by hound CI

Celoka committed:
chore(file delete): remove .env

- remove .env file from repository

Celoka committed:
chore(webpack): add environment variable in webpack

- include environment variable for firebase in webpack
- remove all dead code

Celoka committed:
chore(refactor):  code refactor

- implement SRP in controllers
- change all variable names to camel case

Celoka committed:
chore(webpack): fix production webpack bug

- fix production webpack bug on heroku

Celoka committed:
chore(webpack):  fix production webpack on heroku

- fix production webpack for heroku build

Celoka committed:
chore(webpack):  fix production webpack on heroku

 - fix production webpack for heroku build

Celoka committed:
feature(phone number): include phone number format

- include phone number format in the input field for phone numbers to guide users enter valid detail as phone number

Celoka committed:
chore(version): version routes

- include version for all routes to indicate the current version

Celoka committed:
chore(flight): documentation

- update read me file

[Finishes ]

Celoka committed:
chore(flight): documentation

- update read me file

[Delivers ]

Celoka committed:
chore(test): test

- update test server side test

Celoka committed:
chore(post message): bug

- fix post message bug

[Finishes]

Celoka committed:
feature(component): refactor

- refactor code to remove too many components
- make app responsive

Celoka committed:
chore(webpack): refactor

- update production webpack settings

Celoka committed:
chore(webpack): refactor

- update production webpack settings

Celoka committed:
chore(test): server test

- update server test after code refactor to reflect the current state of route

Celoka committed:
Merge pull request #57 from Celoka/chore/feedback-implementation

Chore/feedback implementation

Celoka committed:
feature(jwt): implement jwt

- authenticate routes with jwt
- remove firebase onAuthStateChange
- add image background to dashboard

efalayi reviewed (commented):
@Celoka it seems you pushed your node_modules to github

efalayi commented on client/src/Routes.jsx:
```diff
@@ -0,0 +1,41 @@
+import DashBoard from './components/DashBoard.jsx';
+import ResetPassword from './components/ResetPassword.jsx';
+
+const RequireAuth = ({ component: Component, ...rest }) => (
```
Add JSDoc

efalayi commented on client/src/actions/AppActions.js:
```diff
@@ -0,0 +1,277 @@
+    return axios
+      .post('/api/v1/user/signin', signInDetails)
+      .then((response) => {
+        const { token } = response.data;
```
You can also do this `const { token, userDetails } = response.data`

efalayi commented on client/src/actions/AppActions.js:
```diff
@@ -0,0 +1,277 @@
+      .post('/api/v1/user/signin', signInDetails)
+      .then((response) => {
+        const { token } = response.data;
+        const user = response.data.userDetails[0];
```
Let the variable name reflect what is being gotten from `userDetails[0]`

efalayi commented on client/src/actions/AppActions.js:
```diff
@@ -0,0 +1,277 @@
+        const token = response.data.user.stsTokenManager.accessToken;
+        const googleUser = response.data.user;
+        const displayName = response.data.user.displayName;
+        localStorage.setItem('token', JSON.stringify(token));
```
I see this `LocalStorage.setItem('token', JSON.stringify(token))` is used a lot. You could create a separate function that does this.

efalayi commented on client/src/actions/AppActions.js:
```diff
@@ -0,0 +1,277 @@
+      .then((response) => {
+        const { userGroups } = response.data;
+        AppDispatcher.dispatch({
+          actionType: AppConstants.SET_GROUP,
```
By `SET_GROUP` do you mean set just one group?. If you're setting more that one group, it should say `SET_GROUPS`

Celoka committed:
feature(jwt): implement jwt

- add jwt verifify to routes to protect them
- abstract create user and login action into function

[Delivers ]

Celoka committed:
Merge pull request #59 from Celoka/feature/implement-jwt-authentication

Feature/implement jwt authentication

efalayi reviewed (changes requested):
1. Your jsdoc could still use some work. Check this [link](http://usejsdoc.org/)
2. You can also add comments in your css file
3. I also observed you use double and single quotes. Try to make it consistent.
Weldone @Celoka .

Celoka committed:
feature(validation): implement form validation

- ensure all input form fields are validated

[Finishes ]

Celoka committed:
Merge pull request #60 from Celoka/feature/client-side-form-validation

Feature(validation): implement form validation

Celoka committed:
feature(group author): implement group author assignment to group

- assign group creator to group when that user creates a group
- don't clear form on form input field validation failure
- validate user group when a conflict occurs

[Finishes ]

Celoka committed:
Merge pull request #61 from Celoka/feature/assign-group-creators-to-group

feature(group author): implement group author assignment to group

Celoka committed:
feature(ui-ux): ui responsiveness

- re write css to scss and compile with webpack
- make ui responsive and improve ux
- on group create group failure state should not clear

[Finishes ]

Celoka committed:
Merge pull request #62 from Celoka/feature/ui-ux-responsiveness

feature(ui-ux): ui responsiveness

Celoka committed:
bug(webpack): fix webpack

- add webpack version to package json

Celoka committed:
bug(webpack): fix webpack

- add extract text webpack plugin to package json

Celoka committed:
bug(webpack): fix webpack

- update all dependencies

Celoka committed:
bug(webpack): fix webpack

- update all dependencies

Celoka committed:
chore(refactor): refactor code

- refactor code to reflect world class guidlines
- improve on flux architecture implementation
- improve naming convention

Celoka committed:
Merge pull request #63 from Celoka/chore/refactor-code-and-improve-flux-use

chore(refactor): refactor code

Celoka committed:
chore(folder structure): arrange app folder

- put files in the approrpiate directory

Celoka committed:
feature(google): implement google login

- provide form for number update to a first time google user account
- fix google login bug
- refactor to reflect single responsibility principle
- create helper function to help achieve srp
[Finishes ]

Celoka committed:
Merge pull request #64 from Celoka/bug/fix-google-login

Bug/fix google login

Celoka committed:
docs(jsdoc): write jsdoc

- write jsdoc for all function action and method

[Finishes ]

Celoka committed:
Merge pull request #65 from Celoka/chore/jsdoc-documentation

docs(jsdoc): write jsdoc

Celoka committed:
chore(refactor): refactor form validation

- refactor repeated codes

Celoka committed:
chore(test): update client side test

- write more client test for component

Celoka committed:
feature(client test): update client test

- write more component test
- update test for actions
- writes test for store
- place files into appropriate folder
- redesign ui
- fix signup/singin redirection bug

[Finishes ]

Celoka committed:
Merge pull request #66 from Celoka/feature/unit-test

chore/write client side test
